src/basic_fun.cpp

```diff
@@ -676,6 +676,117 @@ namespace lib {
     return newObj;
   }
 
+  BaseGDL* heap_refcount( EnvT* e)
+  {
+    static int DISABLEIx = e->KeywordIx("DISABLE");
+    static int ENABLEIx = e->KeywordIx("ENABLE");
+    static int IS_ENABLEDIx = e->KeywordIx("IS_ENABLED");
+//    trace_me = trace_arg();
+    int nParam=e->NParam();
+    
+	GDLInterpreter* interpreter = e->Interpreter();
+
+    if( nParam == 0) {
+	if( e->KeywordSet(DISABLEIx)) {
+	  EnableGC(false);
+	  }
+	else if( e->KeywordSet(ENABLEIx)) {
+	  EnableGC(true);
+	  interpreter->EnableAllGC();
+	}
+	if(e->KeywordPresent(IS_ENABLEDIx)) 
+	      e->SetKW( IS_ENABLEDIx,
+			new DByteGDL( IsEnabledGC()) );
+	return new DIntGDL( 0);
+	}
+
+    BaseGDL* p = e->GetPar( 0);
+    if( p == NULL)
+      {
+	return new DIntGDL( 0);
+      } 
+
+    DIntGDL* ret = new DIntGDL(p->Dim());
+    Guard<DIntGDL> ret_guard(ret);
+    DType pType = p->Type();
+    SizeT nEl = p->N_Elements();
+    if(pType == GDL_OBJ) {
+	DObjGDL* pObj = static_cast<DObjGDL*>( p);
+	for( SizeT i=0; i<nEl; ++i)
+		(*ret)[ i] = interpreter->RefCountHeapObj( (*pObj)[ i]);
+	if( e->KeywordSet(DISABLEIx) or
+	    e->KeywordSet(ENABLEIx) ) {
+	      bool set = e->KeywordSet(ENABLEIx) ? true: false;
+	      interpreter->EnableGCObj( pObj, set);
+	    }
+	}
+    else {
+      if( pType == GDL_PTR) {
+      
+	  DPtrGDL* pPtr = static_cast<DPtrGDL*>( p);
+	  for( SizeT i=0; i<nEl; ++i)
+		  (*ret)[ i] = interpreter->RefCountHeap( (*pPtr)[ i]);
+	  if( e->KeywordSet(DISABLEIx) or
+	      e->KeywordSet(ENABLEIx) ) {
+		bool set = e->KeywordSet(ENABLEIx) ? true: false;
+//	if(trace_me) cout <<" GC set? "<<set<<endl;
```

Comment:
I guess worth cleaning up?

Reply:
trace_me is a debug flag that can easily be activated for future development, the outputs are 
where I was interested at one time and would be again when more effort is given.